### PR TITLE
Fix purge relation

### PR DESCRIPTION
**Pull request by stonebuzz**
Same as https://github.com/glpi-project/glpi/pull/15002

but when the relation is purged

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

File: src/Computer_Item.php [hunk 1]
```diff
@@ -190,6 +190,8 @@ public function cleanDBonPurge()
             $computer = new Computer();
             $computer->getFromDB($this->fields['computers_id']);
 
+            $is_mainitem_dynamic = (bool) ($computer->fields['is_dynamic'] ?? false);
```
Comment (trasher) [inline src/Computer_Item.php:193]: Shoudl we really have a main item not dynamic during inventory?

File: tests/functional/Glpi/Inventory/Assets/Computer.php [hunk 2]
```diff
@@ -192,6 +192,142 @@ public function testPrepare($xml, $asset)
...
+
+        //load manual monitor
+        $manual_monitor = new \Monitor();
+        $this->boolean($manual_monitor->getFromDB(reset($manual_monitors)['items_id']))->isTrue();
+        //check same users
+        $this->integer($manual_monitor->fields['users_id'])->isIdenticalTo($computer->fields['users_id']);
+        $this->integer($manual_monitor->fields['users_id'])->isIdenticalTo(getItemByTypeName('User', 'tech', true));
+
+
+        //Enable option to propagate users_id on update to connected element
```
Comment (trasher) [inline tests/functional/Glpi/Inventory/Assets/Computer.php:298]: I guess this comment is wrong

File: src/Computer_Item.php [hunk 3]
```diff
@@ -226,6 +228,10 @@ public function cleanDBonPurge()
                         }
 
                         if (count($updates)) {
+                            //propage is_dynamic value if needed to prevent locked fields
```
Comment (trasher) [inline src/Computer_Item.php:231]: "propagate" rather than "propage"?

File: src/Computer_Item.php (hunk 1 above)
Comment (stonebuzz) [reply src/Computer_Item.php:193]: I suppose the following items can be added manually

```
$CFG_GLPI["directconnect_types"]          = ['Monitor', 'Peripheral', 'Phone', 'Printer'];
```

File: tests/functional/Glpi/Inventory/Assets/Computer.php (hunk 2 above)
Comment (stonebuzz) [reply tests/functional/Glpi/Inventory/Assets/Computer.php:298]: fixed

File: src/Computer_Item.php (hunk 3 above)
Comment (stonebuzz) [reply src/Computer_Item.php:231]: fixed

File: src/Computer_Item.php (hunk 1 above)
Comment (trasher) [reply src/Computer_Item.php:193]: It's not a blocker anyways

**Review by trasher — APPROVED** [review]

**Pull request merged**